static/app/components/buttonClick.tsx

```diff
@@ -0,0 +1,55 @@
+import {useCallback} from 'react';
+
+import HookStore from 'sentry/stores/hookStore';
+
+const useGetHook = ({disabled, busy, onClick}) => {
+  const useButtonTracking = HookStore.get('react-hook:track-buttons')[0];
+  const useButtonClick = () =>
+    useCallback((e: React.MouseEvent) => {
+      // Don't allow clicks when disabled or busy
+      if (disabled || busy) {
+        e.preventDefault();
+        e.stopPropagation();
+        return;
+      }
+
+      if (typeof onClick !== 'function') {
+        return;
+      }
+
+      onClick(e);
+    }, []);
+
+  const handleClick = useButtonTracking || useButtonClick;
+
+  return handleClick;
+};
+
+export function useButtonHook({
```

Comment:
@roggenkemper If this is the only export, we should use `export default` here and rename the file `useButtonHook`